src/ULMFiT/train_text_classifier.jl

```diff
@@ -0,0 +1,213 @@
+"""
+ULMFiT - Text Classifier
+
+This is wrapper around the LanguageMode struct. It has three fields:
+
+vocab           : contains the same vocabulary from the LanguageModel
+rnn_layers      : contains same DroppedEmebeddings, LSTM (AWD_LSTM) and VarDrop layers of LanguageModel except for last softmax layer
+linear_layers   : contains Chain of two Dense layers [PooledDense and Dense] with softmax layer
+
+To train create and instance and give it as first argument to 'train_classifier!' function
+"""
+mutable struct TextClassifier
+    vocab::Vector
+    rnn_layers::Flux.Chain
+    linear_layers::Flux.Chain
+end
+
+function TextClassifier(lm::LanguageModel=LanguageModel(), clsfr_out_sz::Integer=1, clsfr_hidden_sz::Integer=50, clsfr_hidden_drop::Float64=0.4)
+    return TextClassifier(
+        lm.vocab,
+        lm.layers[1:8],
+        Chain(
+            gpu(PooledDense(length(lm.layers[7].layer.cell.h), clsfr_hidden_sz)),
+            gpu(BatchNorm(clsfr_hidden_sz, relu)),
+            Dropout(clsfr_hidden_drop),
+            gpu(Dense(clsfr_hidden_sz, clsfr_out_sz)),
+            gpu(BatchNorm(clsfr_out_sz)),
+            softmax
+        )
+    )
+end
+
+Flux.@treelike TextClassifier
+
+"""
+Cross Validate
+
+This function will be used to cross-validate the classifier
+
+Arguments:
+
+tc              : Instance of TextClassfier
+gen             : 'Channel' to get a mini-batch from validation set
+num_of_batches  : specifies the number of batches the validation will be done
+
+If num_of_batches is not specified then all the batches which can be given by the
+gen will be used for validation
+"""
+function validate(tc::TextClassifier, gen::Channel, num_of_batches::Union{Colon, Integer})
+    n_classes = size(tc.linear_layers[end-2].W, 1)
+    classifier = mapleaves(Tracker.data, tc)
+    Flux.testmode!(classifier)
+    loss = 0
+    iters = take!(gen)
+    (num_of_batches != : & num_of_batches < iters) && (iters = num_of_batches)
```

Comment:
seems like a typo here. this line does not compile!

Reply:
fixed that